packages/server/lib/middleware/ratelimit.middleware.ts

```diff
@@ -43,11 +42,16 @@ export const rateLimiterMiddleware = (req: Request, res: Response, next: NextFun
             setXRateLimitHeaders(rateLimiterRes);
             next();
         })
-        .catch((rateLimiterRes) => {
-            res.setHeader('Retry-After', Math.floor(rateLimiterRes.msBeforeNext / 1000));
-            setXRateLimitHeaders(rateLimiterRes);
-            logger.info(`Rate limit exceeded for ${key}. Request: ${req.method} ${req.path})`);
-            res.status(429).send('Too Many Requests');
+        .catch((rateLimiterRes: unknown) => {
+            if (rateLimiterRes instanceof RateLimiterRes) {
+                res.setHeader('Retry-After', Math.floor(rateLimiterRes.msBeforeNext / 1000));
+                setXRateLimitHeaders(rateLimiterRes);
+                logger.info(`Rate limit exceeded for ${key}. Request: ${req.method} ${req.path})`);
+                res.status(429).send('Too Many Requests');
+                return;
+            }
+
+            res.status(500).send('Too Many Requests');
```

Comment:
should it be `429: Too Manu requests` or `500: some other error`?

Reply:
oops yes thanks